polyfill/lib/monthday.mjs

```diff
@@ -46,13 +58,18 @@ export class MonthDay {
       const val2 = GetSlot(other, slot);
       if (val1 !== val2) return false;
     }
-    return true;
+    return GetSlot(this, CALENDAR).id === GetSlot(other, CALENDAR).id;
   }
   toString() {
     if (!ES.IsTemporalMonthDay(this)) throw new TypeError('invalid receiver');
     let month = ES.ISODateTimePartString(GetSlot(this, ISO_MONTH));
     let day = ES.ISODateTimePartString(GetSlot(this, ISO_DAY));
     let resultString = `${month}-${day}`;
+    const calendarId = GetSlot(this, CALENDAR).id;
+    if (calendarId !== 'iso8601') {
+      const year = ES.ISOYearString(GetSlot(this, REF_ISO_YEAR));
+      resultString = `${year}-${resultString}[c=${calendarId}]`;
```

Comment:
Optional: It might be nice to pull out the `[c=...]` stuff into a new function (abstract operation).